cdap-relational-ext-calcite/src/main/java/io/cdap/cdap/etl/api/relational/CalciteSqlDialectConverter.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.etl.api.relational;
+
+import io.cdap.cdap.api.data.schema.Schema;
+import org.apache.calcite.prepare.PlannerImpl;
+import org.apache.calcite.schema.SchemaPlus;
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.apache.calcite.sql.parser.SqlParser;
+import org.apache.calcite.sql.validate.SqlConformance;
+import org.apache.calcite.sql.validate.SqlConformanceEnum;
+import org.apache.calcite.sql.validate.SqlValidator;
+import org.apache.calcite.tools.FrameworkConfig;
+import org.apache.calcite.tools.Frameworks;
+import org.apache.calcite.tools.ValidationException;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+
+public class CalciteSqlDialectConverter implements SqlDialectConverter {
```

Comment:
As discussed, it would be great to have tests for the conversion between different dialects.

Reply:
I have validation tests now, but conversion tests will probably be difficult as Calcite tends to add newlines and change cases for the input given, making it difficult to come up with an expected string to compare with.